lib/seattleflu/db/cli/command/__init__.py

```diff
@@ -9,3 +10,12 @@
     "clinical",
     "user",
 ]
+
+
+def trim_whitespace(df: pd.DataFrame) -> pd.DataFrame:
+    """ Trims leading and trailing whitespace from strings in *df* """
+    str_dtypes = df.select_dtypes(include='object')
+
+    df[str_dtypes.columns] = str_dtypes.apply(lambda column: column.str.strip())
```

Comment:
The `object` dtype also catches things which aren't strings, such as if we had sets or lists or other objects in a column. When `column.str.strip()` is applied to a object-but-non-string column, NaNs are returned. For example:

```python
>>> d = pd.DataFrame({"a":["one","two  ", "  three"], "b":[{1,2,3},{3,4,5},{7,8,9}], "c":[1,2,3]})
>>> d
         a          b  c
0      one  {1, 2, 3}  1
1    two    {3, 4, 5}  2
2    three  {8, 9, 7}  3
>>> d.dtypes
a    object
b    object
c     int64
dtype: object
>>> d.select_dtypes(include = 'object').apply(lambda c: c.str.strip())
       a   b
0    one NaN
1    two NaN
2  three NaN
```

It would be more robust and more defensive to filter the selected columns to just those which are a subclass of `str`. I think something like the following would do it:

```python
str_columns = df.columns[df.applymap(lambda x: isinstance(x, str) or pd.isna(x)).all()]

df[str_columns] = df[str_columns].apply(lambda column: column.str.strip())
```

Reply:
Good catch! This was something I noticed in the docs and ran into when implementing, but I (dangerously) assumed we were 'safe' from having to worry about this.